### DISCUSSION TITLE
How to update the dom when user redirect to other page via hx-post or hx-get?

### DISCUSSION THREAD

**Discussion by Junaidiqbal35** [bigskysoftware/htmx#1750] [Q&A]
Hi, I'm a django dev..  I'm facing some problem in htmx redirect.  
Just take as an of login page if you are using hx-post for login purpose, how will you update the dom when user redirect to other page? I tried hx-swap='outerHTML' but on redirect page, login form still show.  maybe I'm missing something. Anyone who want to share their experience in this?

**Comment by Telroshan** [comment]
Hey, I'm not sure I understand the question, I'm assuming you're talking about a login form, that should redirect your user on a successful authentication, but the redirect seems weird and didn't update the whole page (let me know if that's a misinterpretation there)

So first, know that using a [standard HTTP redirect](https://developer.mozilla.org/docs/Web/HTTP/Redirections) will probably be unintuitive at first:
You might expect your client browser to take the user to a completely new page with the URL redirected to, but this it not what happens with htmx.
As it's a partial request (compared to a standard form that would take the user to another page when submitting it), when you do a standard redirect:
- htmx follows the redirection
- htmx gets the response of that redirected URL
- htmx swaps it in place of your initial hx-target/swap strategy

This is actually the expected behaviour, as you might want to redirect partial requests and swap the content of something else than the user asked for in the first place, without redirecting the whole page.
So if you're using a standard redirect on the login page, you will actually _stay_ on that login page, and not take the user to a new one.
Note that htmx cannot know without additional information, whether you want a full or partial redirect in that case.

**Solution**: this is where the [HX-Redirect](https://htmx.org/reference/#response_headers) header comes in, which instructs htmx to do a full page reload instead. Sending that header on a successful login for example, will bring your user to another whole page
